### PR TITLE
Base+LibGUI+LibGfx+WindowServer: Add Redmond 10 theme

### PR DESCRIPTION
**Pull request by metalvoidzz**
You surely love a good flat theme, right? If so, this might be for you:

redmond 10

To make this theme work, I had to fix up some issues I found along the way:

First up, zero-size window borders could not be dragged by the mouse.

Second, the style painter used to draw sliders and progress bars using the window border colors. This worked fine until now, but our beloved commercial OS that inspired this theme makes heavy use of the color white, which resulted in a white->white gradient on a white background.

### CODE REVIEW

**Comment by linusg** [comment]
> First up, zero-size window borders could not be dragged by the mouse.

Windows 10 has a 1px border. :)

**Comment by filiphsandstrom** [comment]
This looks amazing 💯

File: Base/res/themes/Redmond 10.ini [hunk 1]
```diff
@@ -0,0 +1,108 @@
+[Metrics]
+TitleButtonWidth=44
+BorderRadius=0
+TitleButtonHeight=24
+TitleHeight=28
+BorderThickness=0
```
Comment (linusg) [inline Base/res/themes/Redmond 10.ini:6]: See previous comment

Comment (metalvoidzz) [reply Base/res/themes/Redmond 10.ini]: fixed :^)

**Comment by iProgramMC** [comment]
> > First up, zero-size window borders could not be dragged by the mouse.
> 
> Windows 10 has a 1px border. :)



> > First up, zero-size window borders could not be dragged by the mouse.
> 
> Windows 10 has a 1px border. :)

Technically it just has an invisible bigger border - this is why you're able to drag the window to resize even if you aren't perfectly on the pixel. It's a DWM hack. I think your `rect_adjusted_for_hit_test` should check for a theme property or something

**Comment by 2xsaiko** [comment]
If possible (I haven't looked at how themes work yet), I'd inset the icon towards the right by the same distance as the margins at the top/bottom of the icon in the title bar so that it doesn't feel as close to the edge, and maybe also add the same amount of right margin. So like this terrible mockup (current one is top left)

image

Looks great otherwise though! (though I prefer the classic style :^) )

**Comment by djwisdom** [comment]
## Suggested tweak on TitleButtonIcons :

https://user-images.githubusercontent.com/391659/172782264-0d6f4364-bbe3-4ca0-ac65-a9b3349b4a2f.mp4

**Comment by metalvoidzz** [comment]
@djwisdom That looks pretty cool, but I found some issues while trying to do the same.
AFAIK there's currently no way to give icons a background color, so I had to resize them to 45x29, which does not work for notification frames:
issue2
Also, ClassicWindowTheme leaves a 2-pixel border around the close icon, while Windows 10 has none:
issue1
Removing that border breaks most other themes unfortunately.

**Comment by djwisdom** [comment]
Well, for a Win10 theme, yours is currently holding the best looking composition imho. I'm impressed with the effort you put in. And the result says it all.

**Pull request closed without merging**